bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java

```diff
@@ -587,13 +604,21 @@ public boolean hostNameAndIpMappingExists(long addressObjectId) throws TskCoreEx
 	 * @param hostAddress The host address.
 	 */
 	public void addUsage(Content content, HostAddress hostAddress) throws TskCoreException {
+		
+		String key = content.getDataSource().getId() + "#" + hostAddress.getId();
+		Long cachedValue = hostAddressUsageCache.getIfPresent(key);
+		if (null != cachedValue && cachedValue == content.getId()) {
```

Comment:
Is there any reason to do the second part of this check?

Reply:
@APriestman Yes; usage is associated with both a data source as well as the content. The key is on datasourceId.

Reply:
Ok I see that now, but then why wouldn't the key be on all three fields? Isn't the point just to check if you've already added (hostAddrId, contentId, datasourceId) to the database?

Reply:
I have updated it with all 3 now